create.go

```
@@ -18,9 +18,17 @@ func (r Repository[M, D, SF, SORD, SO, UF]) Create(ctx context.Context, model M)
 	r.logDebugf(actionCreate, "data: %+v", data)
 
+	// Use the injected tracer
+	ctx, seg := r.tracer.BeginSubSegment(ctx, fmt.Sprintf("MongoDB.InsertOne.%s", r.collectionName))
+	defer func() {
+		seg.Close(nil)
+	}()
+
 	res, err := r.Collection().InsertOne(ctx, data)
 	if err != nil {
 		r.logErrorf(err, actionCreate, "error inserting %s DAO", r.collectionName)
+		seg.AddError(err)
```

golangci-lint (1.64.8)

Error return value of `seg.AddError` is not checked

(errcheck)